### ISSUE TITLE
dynamic query

### ISSUE THREAD

**Issue by Viper-Bit** [isar/isar#431]
**Steps to Reproduce**
when i run my dynamic query with exceptId or synced param flutter crashes without any error or exception (for calling isar_filter_not)
i dont know its a bug or my dynamic query is wrong

**Code sample**
```dart
import 'package:isar/isar.dart';

part 'tag.g.dart';

@Collection()
class Tag {
  @Id()
  int? isarId;

  @Index(unique: true, replace: true)
  late String id;

  late String name;
  late bool synced;

  static Future<List<Tag>> query(String pattern,
      [List<String>? exceptId, bool? synced]) async {
    final searchList = pattern.split(' ');
    var patternFilter = <FilterCondition>[];

    for (var i = 0; i <= searchList.length - 1; ++i) {
      if (searchList[i] != '') {
        patternFilter.add(FilterCondition.startsWith(
          property: 'name',
          value: searchList[i],
          caseSensitive: false,
        ));
      }
    }

    return isar.tags.buildQuery<Tag>(
      filter: FilterGroup.and([
        FilterGroup.or(patternFilter),
        if (exceptId?.isNotEmpty ?? false)
          FilterGroup.not(FilterGroup.or(exceptId!.map((id) {
            return FilterCondition.equalTo(
              property: 'id',
              value: id,
            );
          }).toList())),
        if (synced != null)
          FilterCondition.equalTo(
            property: 'synced',
            value: synced,
          ),
      ]),
      sortBy: const [SortProperty(property: 'name', sort: Sort.asc)],
    ).findAll();
  }
}

```

**Version**
 - Platform: Windows
 - Flutter version: 3.1.0-9.0.pre
 - Isar version: 3.0.0-dev.1
 - Isar Generator version: 3.0.0-dev.1

**Comment by leisim** [comment]
I think there is already a fix in the dev branch. I'll release it shortly :)

**Comment by Viper-Bit** [comment]
@leisim thx hope to see new version soon, right now im using optional and anyof

**Comment by Viper-Bit** [comment]
fixed in 3.0.0-dev.2
